src/core/tsi/ssl_transport_security_utils.cc

```diff
@@ -247,4 +248,40 @@ tsi_result SslProtectorUnprotect(const unsigned char* protected_frames_bytes,
   return result;
 }
 
+int verify_crl_signature(X509_CRL* crl, X509* issuer) {
+  EVP_PKEY* ikey = X509_get_pubkey(issuer);
+  if (ikey == nullptr) {
+    // Can't verify signature because we couldn't get the pubkey, fail the
+    // check.
+    EVP_PKEY_free(ikey);
+    return 1;
+  }
+  int ret = X509_CRL_verify(crl, ikey);
+  EVP_PKEY_free(ikey);
+  return ret;
+}
+
+int verify_crl_cert_issuer_names_match(X509_CRL* crl, X509* issuer) {
+  return X509_NAME_cmp(X509_get_issuer_name(issuer), X509_CRL_get_issuer(crl));
```

Comment:
Should we be using `X509_get_subject_name(issuer)` instead of `X509_get_issuer_name(issuer)` here?

Reply:
I don't think so, but it depends on how we call it

In this case, we're asking whether a given certificate and a given crl have the same issuer. Because a CRL issued by issuer A would be the right CRL to revoke a cert issued by issuer A. But a cert issued by issuer A would have a different subject.
To check the subject name against the CRL we'd have to go one up the chain, because with Direct CRLs the subject one up the chain is the issuer.

Does that make sense to you?

Reply:
OK so to confirm, what we want to do here is: given a cert and CRL, do they have the same issuer. Is that correct?

If yes, then I think the problem is using `issuer` as the variable name rather than e.g. `cert`?

WDYT?

Reply:
Ah that makes a lot of sense, I've fixed it